crates/re_viewer/src/app.rs

```diff
@@ -934,14 +940,43 @@ impl App {
                     if let Some(err) = err {
                         re_log::warn!("Data source {} has left unexpectedly: {err}", msg.source);
                     } else {
-                        re_log::debug!("Data source {} has left", msg.source);
+                        re_log::debug!("Data source {} has finished", msg.source);
+
+                        // This could be the signal that we finished loading a blueprint.
+                        // In that case, we want to make it the default.
+
+                        if let Some(entity_db) =
+                            store_hub.entity_db_from_channel_source(&channel_source)
+                        {
+                            if let Some(store_info) = entity_db.store_info() {
+                                match store_info.store_id.kind {
+                                    StoreKind::Recording => {
+                                        // Recordings become active as soon as we start streaming them.
+                                    }
+                                    StoreKind::Blueprint => {
+                                        // We wait with activaing blueprints until they are fully loaded,
+                                        // so that we don't run heuristics on half-loaded blueprints.
+                                        re_log::debug!("Activating newly loaded blueprint");
+                                        store_hub.set_blueprint_for_app_id(
+                                            entity_db.store_id().clone(),
+                                            store_info.application_id.clone(),
+                                        );
```

Comment:
While this is definitely an improvement for the file-open use-case, moving this here regresses the current blueprint-on-connect behavior.

Because blueprint and recording share the same channel, the blueprint now won't be activated until the channel is closed.

You can see this issue by adding:
```
import time
time.sleep(5)
```
to the end of `examples/python/blueprint/main.py`

This causes the saved blueprint to load first. And then after the timeout when the whole connection is closed, the blueprint loads.

This alone is insufficient for https://github.com/rerun-io/rerun/issues/5297... we still need some kind of safe-to-load marker inside the stream.

Reply:
ouch; good point…

Reply:
Yeah, I might merge this and then start work on https://github.com/rerun-io/rerun/issues/5297